src/audio/tone.c

```diff
@@ -530,7 +530,7 @@ static int tone_cmd_set_data(struct comp_dev *dev,
 	case SOF_CTRL_CMD_ENUM:
 		comp_info(dev, "tone_cmd_set_data(), SOF_CTRL_CMD_ENUM, cdata->index = %u",
 			  cdata->index);
-		compv = (struct sof_ipc_ctrl_value_comp *)cdata->data->data;
+		compv = ASSUME_ALIGNED(cdata->data->data, 4);
```

Comment:
shouldnt this be `COMP_GET_IPC` also shouldnt we update that macro?

Reply:
yes, we should and XCC seems to complain on certain configs too.

Reply:
> shouldnt this be `COMP_GET_IPC` also shouldnt we update that macro?

I think no, this is sof_ipc_ctrl_value_comp, not the sof_ipc_comp.

@lgirdwood looks the compiler will complain without typecasting, to align with other place, maybe use this:
```
compv = (struct sof_ipc_ctrl_value_comp *)ASSUME_ALIGNED(cdata->data->data, 4);
```

And, we might need similar change for detect_test also: https://github.com/thesofproject/sof/blob/d9a7dfd2316067d6d5d1268d7f7360f4f0cbe81a/src/samples/audio/detect_test.c#L430

Reply:
ah, yes - it's IPC private data specific to the use case. Let me fix this, thanks @keyonjie and @cujomalainey